compiler/rustc_typeck/src/coherence/builtin.rs

```diff
@@ -342,7 +344,7 @@ pub fn coerce_unsized_info<'tcx>(tcx: TyCtxt<'tcx>, impl_did: DefId) -> CoerceUn
     let param_env = tcx.param_env(impl_did);
     assert!(!source.has_escaping_bound_vars());
 
-    let err_info = CoerceUnsizedInfo { custom_kind: None };
+    let err_info = CoerceUnsizedKind::Ptr;
```

Comment:
Do we want to return something like `None` (i.e. make this function return an `Option` or `Result`? I think `CoerceUnsizedInfo { custom_kind: None }` was used here as both like `CoerceUnsizedKind::Ptr` _and_ the "uhh i don't know" coercion kind. 

Either that, or we should make a `CoerceUnsizedKind::Error`, though I'm not a fan of that.

Reply:
Like it seems the code below is returning `err_info` on the error path (e.g. when we try to coerce two ADTs with different `def_id`s).

Reply:
First of all, this PR definitely doesn't make the situation *worse*, since the same amount of information is produced. It seems at least *somewhat* consistent to "treat this as if a builtin pointer cast" (for the purpose of continuing reporting errors) if the coercion is bad (since we won't get to codegen anyway).

It's also worth noting where this is actually used, which seems to always be for side effects or go through `custom_coerce_unsize_info` in practice, and `CoerceUnsizedKind::Ptr` is only used as an error in `find_vtable_types_for_unsizing` anyway